src/utils.rs

```diff
@@ -252,7 +258,9 @@ pub(crate) fn open_database(
     wallet_opts: &WalletOpts,
     home_path: &Path,
 ) -> Result<AnyDatabase, Error> {
+    #![allow(unused_variables)]
     let wallet_name = wallet_opts.wallet.as_ref().expect("wallet name");
+    #[cfg(any(feature = "key-value-db", feature = "sqlite-db",))]
     let database_path = prepare_wallet_db_dir(wallet_name, home_path)?;
```

Comment:
The `home_path` is also unused here in case of memory db. So thats triggering another unused warning.. Why not just mark the whole function with `#[allow(unused_variables)]`. That takes care of all the unused..

Reply:
Oh because you are using an inner attribute.. Just do outer attribute like  `#[allow(unused_variable)]` without the `!` mark.. That should work..

Reply:
Oh that's right, thanks! I just updated the code :)